File: Systems-for-Design-and-Implementation/Exam/Schita/ExamenMPP/ExamenMPP/Networking/src/main/java/app/network/jsonprotocol/Response.java
package app.network.jsonprotocol;

// TODO 1: IMPORT MODELS FOR RESPONSE CLASS

public class Response {

    private ResponseType type;
    private String errorMessage;

    // TODO 2: ADD MODELS FOR RESPONSE CLASS

    public Response(){}

    public ResponseType getType() {
        return type;
    }

    public void setType(ResponseType type) {
        this.type = type;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }

    /* TODO 3: CREATE toString() FOR RESPONSE CLASS
    @Override
    public String toString() {
        return "Response{" +
                "type=" + type +
                ", reservationManager=" + reservationManager +
                ", trip=" + trip +
                ", reservation=" + reservation +
                ", seat=" + seat +
                '}';
     */
}

File: S4-Systems-for-Design-and-Impl/Labs/Lab9/ProiectCSharp/TransportNetworking/transport.network/jsonprotocol/Response.cs
using System;
using System.Collections.Generic;
using Transport.Model.transport.model;
using TransportModel.transport.model;
using TransportNetworking.transport.network.jsonprotocol;

namespace TransportNetworking.transport.network.jsonprotocol
{
    [Serializable]
    public class Response
    {
        public ResponseType type { get; set; }
        public string errorMessage { get; set; }
        public ReservationManager ReservationManager { get; set; }
        public Trip trip { get; set; }
        public Seat Seat { get; set; }
        public Reservation reservation { get; set; }
        public Trip[] trips { get; set; }
        public List<Reservation> reservations { get; set; }
        public List<Seat> seats { get; set; }

        public override string ToString()
        {
            return $"Response{{type={type}, reservationManager={ReservationManager}, trip={trip}, reservation={reservation}, seat={Seat}}}";
        }
    }
}